types/options.go

```diff
@@ -305,7 +305,11 @@ func getRootlessStorageOpts(rootlessUID int, systemOpts StoreOptions) (StoreOpti
 	}
 	if opts.GraphDriverName == "" {
 		if len(systemOpts.GraphDriverPriority) == 0 {
-			opts.GraphDriverName = "vfs"
+			if canUseRootlessOverlay(opts.GraphRoot, opts.RunRoot) {
```

Comment:
Should this check if last usage is already using `vfs` ? I think can be verified by checking if `GraphRoot` already contains a `vfs` directory or not. If yes then assume user is already using `vfs` and do not switch to overlay ?